### ISSUE TITLE
[Core] Dashboard error on ray.init since 2.8.0

### ISSUE THREAD

**Issue by MonkeyCMonkeyD0** [ray-project/ray#41354]
#### 1. the bug

Since update 2.8 of ray (RLlib in my case but error coming from ray-core) I have this error with the dashboard blocking everything.

Here is the displayed error from `ray.init`:
```
2023-11-23 03:55:25,517 ERROR services.py:1329 -- Failed to start the dashboard , return code 0
2023-11-23 03:55:25,517 ERROR services.py:1354 -- Error should be written to 'dashboard.log' or 'dashboard.err'. We are printing the last 20 lines for you. See 'https://docs.ray.io/en/master/ray-observability/ray-logging.html#logging-directory-structure' to find where the log file is.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 103, in wrapper
    return func(*args, **kwargs)
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/_private/worker.py", line 1567, in init
    _global_node = ray._private.node.Node(
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/_private/node.py", line 307, in __init__
    self.start_head_processes()
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/_private/node.py", line 1324, in start_head_processes
    self.start_api_server(
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/_private/node.py", line 1073, in start_api_server
    self._webui_url, process_info = ray._private.services.start_api_server(
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/_private/services.py", line 1396, in start_api_server
    raise e from e
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/_private/services.py", line 1384, in start_api_server
    raise Exception(last_log_str)
Exception: 
The last 20 lines of /tmp/ray/session_2023-11-23_03-55-23_453022_7830/logs/dashboard.log (it contains the error message from the dashboard): 
2023-11-23 03:55:25,355 WARNING dashboard.py:257 -- The dashboard on node workstation failed with the following error:
Traceback (most recent call last):
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/dashboard.py", line 248, in <module>
    loop.run_until_complete(dashboard.run())
  File "/home/user/micromamba/envs/test-env/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/dashboard.py", line 75, in run
    await self.dashboard_head.run()
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/head.py", line 330, in run
    self.http_server = await self._configure_http_server(modules)
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/head.py", line 158, in _configure_http_server
    http_server = HttpServerDashboardHead(
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/http_server_head.py", line 89, in __init__
    raise ex
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/http_server_head.py", line 80, in __init__
    build_dir = setup_static_dir()
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/http_server_head.py", line 38, in setup_static_dir
    raise dashboard_utils.FrontendNotFoundError(
ray.dashboard.utils.FrontendNotFoundError: [Errno 2] Dashboard build directory not found. If installing from source, please follow the additional steps required to build the dashboard(cd python/ray/dashboard/client && npm ci && npm run build): '/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/client/build'
```

#### 2. expected behavior
The dashboard should start correctly without any error when the library is initialized.

#### 3. useful information
Here is the corresponding log `(/tmp/ray/session_2023-11-23_03-55-23_453022_7830/logs/dashboard.log)`:
```
2023-11-23 03:55:25,094	INFO head.py:150 -- Dashboard head grpc address: 0.0.0.0:36327
2023-11-23 03:55:25,100	INFO head.py:254 -- Starting dashboard metrics server on port 44227
2023-11-23 03:55:25,101	INFO utils.py:112 -- Get all modules by type: DashboardHeadModule
2023-11-23 03:55:25,348	INFO utils.py:145 -- Available modules: [<class 'ray.dashboard.modules.actor.actor_head.ActorHead'>, <class 'ray.dashboard.modules.event.event_head.EventHead'>, <class 'ray.dashboard.modules.healthz.healthz_head.HealthzHead'>, <class 'ray.dashboard.modules.job.job_head.JobHead'>, <class 'ray.dashboard.modules.log.log_head.LogHead'>, <class 'ray.dashboard.modules.metrics.metrics_head.MetricsHead'>, <class 'ray.dashboard.modules.node.node_head.NodeHead'>, <class 'ray.dashboard.modules.reporter.reporter_head.ReportHead'>, <class 'ray.dashboard.modules.serve.serve_rest_api_impl.create_serve_rest_api.<locals>.ServeRestApiImpl'>, <class 'ray.dashboard.modules.snapshot.snapshot_head.APIHead'>, <class 'ray.dashboard.modules.state.state_head.StateHead'>, <class 'ray.dashboard.modules.usage_stats.usage_stats_head.UsageStatsHead'>]
2023-11-23 03:55:25,348	INFO head.py:223 -- Modules to load: {'ServeRestApiImpl', 'JobHead', 'NodeHead', 'ActorHead', 'UsageStatsHead', 'StateHead', 'ReportHead', 'APIHead', 'MetricsHead', 'LogHead', 'HealthzHead', 'EventHead'}
2023-11-23 03:55:25,349	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.actor.actor_head.ActorHead'>
2023-11-23 03:55:25,349	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.event.event_head.EventHead'>
2023-11-23 03:55:25,349	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.healthz.healthz_head.HealthzHead'>
2023-11-23 03:55:25,349	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.job.job_head.JobHead'>
2023-11-23 03:55:25,349	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.log.log_head.LogHead'>
2023-11-23 03:55:25,353	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.metrics.metrics_head.MetricsHead'>
2023-11-23 03:55:25,353	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.node.node_head.NodeHead'>
2023-11-23 03:55:25,353	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.reporter.reporter_head.ReportHead'>
2023-11-23 03:55:25,353	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.serve.serve_rest_api_impl.create_serve_rest_api.<locals>.ServeRestApiImpl'>
2023-11-23 03:55:25,353	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.snapshot.snapshot_head.APIHead'>
2023-11-23 03:55:25,353	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.state.state_head.StateHead'>
2023-11-23 03:55:25,354	INFO head.py:226 -- Loading DashboardHeadModule: <class 'ray.dashboard.modules.usage_stats.usage_stats_head.UsageStatsHead'>
2023-11-23 03:55:25,354	INFO head.py:239 -- Loaded 12 modules. [<ray.dashboard.modules.actor.actor_head.ActorHead object at 0x7f615a50da20>, <ray.dashboard.modules.event.event_head.EventHead object at 0x7f615a50d210>, <ray.dashboard.modules.healthz.healthz_head.HealthzHead object at 0x7f615a50da50>, <ray.dashboard.modules.job.job_head.JobHead object at 0x7f6149ed4970>, <ray.dashboard.modules.log.log_head.LogHead object at 0x7f6149ed49d0>, <ray.dashboard.modules.metrics.metrics_head.MetricsHead object at 0x7f6149ed4a60>, <ray.dashboard.modules.node.node_head.NodeHead object at 0x7f6149ed4eb0>, <ray.dashboard.modules.reporter.reporter_head.ReportHead object at 0x7f6149ed4ee0>, <ray.dashboard.modules.serve.serve_rest_api_impl.create_serve_rest_api.<locals>.ServeRestApiImpl object at 0x7f6149ed4f10>, <ray.dashboard.modules.snapshot.snapshot_head.APIHead object at 0x7f6149ed51e0>, <ray.dashboard.modules.state.state_head.StateHead object at 0x7f6149ed5210>, <ray.dashboard.modules.usage_stats.usage_stats_head.UsageStatsHead object at 0x7f6149ed5300>]
2023-11-23 03:55:25,354	INFO head.py:329 -- Initialize the http server.
2023-11-23 03:55:25,354	ERROR events.py:80 -- Error in grpc_server_request_call: 12
NoneType: None
2023-11-23 03:55:25,355	ERROR base_events.py:1758 -- Task exception was never retrieved
future: <Task finished name='Task-3' coro=<<coroutine without __name__>()> exception=InternalError('Error in grpc_server_request_call: 12')>
Traceback (most recent call last):
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 1037, in shutdown
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 1041, in grpc._cython.cygrpc.AioServer.shutdown
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 1021, in _start_shutting_down
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 959, in _server_main_loop
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 939, in _request_call
grpc._cython.cygrpc.InternalError: Error in grpc_server_request_call: 12
2023-11-23 03:55:25,355	WARNING dashboard.py:257 -- The dashboard on node workstation failed with the following error:
Traceback (most recent call last):
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/dashboard.py", line 248, in <module>
    loop.run_until_complete(dashboard.run())
  File "/home/user/micromamba/envs/test-env/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/dashboard.py", line 75, in run
    await self.dashboard_head.run()
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/head.py", line 330, in run
    self.http_server = await self._configure_http_server(modules)
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/head.py", line 158, in _configure_http_server
    http_server = HttpServerDashboardHead(
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/http_server_head.py", line 89, in __init__
    raise ex
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/http_server_head.py", line 80, in __init__
    build_dir = setup_static_dir()
  File "/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/http_server_head.py", line 38, in setup_static_dir
    raise dashboard_utils.FrontendNotFoundError(
ray.dashboard.utils.FrontendNotFoundError: [Errno 2] Dashboard build directory not found. If installing from source, please follow the additional steps required to build the dashboard(cd python/ray/dashboard/client && npm ci && npm run build): '/home/user/micromamba/envs/test-env/lib/python3.10/site-packages/ray/dashboard/client/build'
```

### Versions / Dependencies

OS: `PRETTY_NAME="Ubuntu 22.04.3 LTS"`
micromamba: `1.5.3`
python: `Python 3.10.13 | packaged by conda-forge | (main, Oct 26 2023, 18:07:37) [GCC 12.3.0] on linux`
ray: 
```
ray-core                   2.8.0       py310h2ca9b2b_1      conda-forge
ray-default                2.8.0       py310hff52083_1      conda-forge
```

other lib in same conda environment: (resulting only from `micromamba install ray-default=2.8`)
```
_libgcc_mutex              0.1         conda_forge          conda-forge
_openmp_mutex              4.5         2_gnu                conda-forge
aiohttp                    3.9.0       py310h2372a71_0      conda-forge
aiohttp-cors               0.7.0       py_0                 conda-forge
aiosignal                  1.3.1       pyhd8ed1ab_0         conda-forge
async-timeout              4.0.3       pyhd8ed1ab_0         conda-forge
attrs                      23.1.0      pyh71513ae_1         conda-forge
blessed                    1.19.1      pyhe4f9e05_2         conda-forge
brotli-python              1.1.0       py310hc6cd4ac_1      conda-forge
bzip2                      1.0.8       hd590300_5           conda-forge
c-ares                     1.22.1      hd590300_0           conda-forge
ca-certificates            2023.11.17  hbcca054_0           conda-forge
cachetools                 5.3.2       pyhd8ed1ab_0         conda-forge
certifi                    2023.11.17  pyhd8ed1ab_0         conda-forge
cffi                       1.16.0      py310h2fee648_0      conda-forge
charset-normalizer         3.3.2       pyhd8ed1ab_0         conda-forge
click                      8.1.7       unix_pyh707e725_0    conda-forge
colorama                   0.4.6       pyhd8ed1ab_0         conda-forge
colorful                   0.5.4       pyhd8ed1ab_0         conda-forge
cryptography               41.0.5      py310h75e40e8_0      conda-forge
distlib                    0.3.7       pyhd8ed1ab_0         conda-forge
filelock                   3.13.1      pyhd8ed1ab_0         conda-forge
frozenlist                 1.4.0       py310h2372a71_1      conda-forge
google-api-core            2.14.0      pyhd8ed1ab_0         conda-forge
google-auth                2.23.4      pyhca7485f_0         conda-forge
googleapis-common-protos   1.61.0      pyhd8ed1ab_0         conda-forge
gpustat                    1.1.1       pyhd8ed1ab_0         conda-forge
grpcio                     1.58.2      py310h1b8f574_0      conda-forge
idna                       3.4         pyhd8ed1ab_0         conda-forge
importlib-metadata         6.8.0       pyha770c72_0         conda-forge
importlib_resources        6.1.1       pyhd8ed1ab_0         conda-forge
jsonschema                 4.20.0      pyhd8ed1ab_0         conda-forge
jsonschema-specifications  2023.11.1   pyhd8ed1ab_0         conda-forge
ld_impl_linux-64           2.40        h41732ed_0           conda-forge
libabseil                  20230802.1  cxx17_h59595ed_0     conda-forge
libblas                    3.9.0       20_linux64_openblas  conda-forge
libcblas                   3.9.0       20_linux64_openblas  conda-forge
libffi                     3.4.2       h7f98852_5           conda-forge
libgcc-ng                  13.2.0      h807b86a_3           conda-forge
libgfortran-ng             13.2.0      h69a702a_3           conda-forge
libgfortran5               13.2.0      ha4646dd_3           conda-forge
libgomp                    13.2.0      h807b86a_3           conda-forge
libgrpc                    1.58.2      he06187c_0           conda-forge
liblapack                  3.9.0       20_linux64_openblas  conda-forge
libnsl                     2.0.1       hd590300_0           conda-forge
libopenblas                0.3.25      pthreads_h413a1c8_0  conda-forge
libprotobuf                4.24.3      hf27288f_1           conda-forge
libre2-11                  2023.06.02  h7a70373_0           conda-forge
libsqlite                  3.44.1      h2797004_0           conda-forge
libstdcxx-ng               13.2.0      h7e041cc_3           conda-forge
libunwind                  1.6.2       h9c3ff4c_0           conda-forge
libuuid                    2.38.1      h0b41bf4_0           conda-forge
libzlib                    1.2.13      hd590300_5           conda-forge
msgpack-python             1.0.6       py310hd41b1e2_0      conda-forge
multidict                  6.0.4       py310h2372a71_1      conda-forge
ncurses                    6.4         h59595ed_2           conda-forge
numpy                      1.26.0      py310hb13e2d6_0      conda-forge
nvidia-ml-py               12.535.133  pyhd8ed1ab_0         conda-forge
opencensus                 0.11.3      pyhd8ed1ab_0         conda-forge
opencensus-context         0.1.3       py310hff52083_2      conda-forge
openssl                    3.1.4       hd590300_0           conda-forge
packaging                  23.2        pyhd8ed1ab_0         conda-forge
pip                        23.3.1      pyhd8ed1ab_0         conda-forge
pkgutil-resolve-name       1.3.10      pyhd8ed1ab_1         conda-forge
platformdirs               3.11.0      pyhd8ed1ab_0         conda-forge
prometheus_client          0.19.0      pyhd8ed1ab_0         conda-forge
protobuf                   4.24.3      py310h620c231_1      conda-forge
psutil                     5.9.5       py310h2372a71_1      conda-forge
py-spy                     0.3.14      h87a5ac0_0           conda-forge
pyasn1                     0.5.0       pyhd8ed1ab_0         conda-forge
pyasn1-modules             0.3.0       pyhd8ed1ab_0         conda-forge
pycparser                  2.21        pyhd8ed1ab_0         conda-forge
pydantic                   1.10.13     py310h2372a71_1      conda-forge
pyopenssl                  23.3.0      pyhd8ed1ab_0         conda-forge
pysocks                    1.7.1       pyha2e5f31_6         conda-forge
python                     3.10.13     hd12c33a_0_cpython   conda-forge
python_abi                 3.10        4_cp310              conda-forge
pyu2f                      0.1.5       pyhd8ed1ab_0         conda-forge
pyyaml                     6.0.1       py310h2372a71_1      conda-forge
ray-core                   2.8.0       py310h2ca9b2b_1      conda-forge
ray-default                2.8.0       py310hff52083_1      conda-forge
re2                        2023.06.02  h2873b5e_0           conda-forge
readline                   8.2         h8228510_1           conda-forge
referencing                0.31.0      pyhd8ed1ab_0         conda-forge
requests                   2.31.0      pyhd8ed1ab_0         conda-forge
rpds-py                    0.13.1      py310hcb5633a_0      conda-forge
rsa                        4.9         pyhd8ed1ab_0         conda-forge
setproctitle               1.3.3       py310h2372a71_0      conda-forge
setuptools                 68.2.2      pyhd8ed1ab_0         conda-forge
six                        1.16.0      pyh6c4a22f_0         conda-forge
smart_open                 6.4.0       pyhd8ed1ab_0         conda-forge
tk                         8.6.13      noxft_h4845f30_101   conda-forge
typing-extensions          4.8.0       hd8ed1ab_0           conda-forge
typing_extensions          4.8.0       pyha770c72_0         conda-forge
tzdata                     2023c       h71feb2d_0           conda-forge
urllib3                    2.1.0       pyhd8ed1ab_0         conda-forge
virtualenv                 20.21.0     pyhd8ed1ab_0         conda-forge
wcwidth                    0.2.12      pyhd8ed1ab_0         conda-forge
wheel                      0.41.3      pyhd8ed1ab_0         conda-forge
xz                         5.2.6       h166bdaf_0           conda-forge
yaml                       0.2.5       h7f98852_2           conda-forge
yarl                       1.9.2       py310h2372a71_1      conda-forge
zipp                       3.17.0      pyhd8ed1ab_0         conda-forge
```

### Reproduction script

```
micromamba create -n test-env -y python=3.10 ray-default=2.8
micromamba run -n test-env python3
import ray; ray.init(include_dashboard=True)
```

### Issue Severity

~~High: It blocks me from completing my task.~~
Medium: It is a significant difficulty but I can work around it.

### Quick Temporary fixe

Install manually the `ray-dashboard` package via:
```
micromamba install -n test-env ray-dashboard
```
then the script execute as:
```python
>>> import ray; ray.init(include_dashboard=True)
2023-11-23 04:25:17,312 INFO worker.py:1664 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265 
RayContext(dashboard_url='127.0.0.1:8265', python_version='3.10.13', ray_version='2.8.0', ray_commit='{{RAY_COMMIT_SHA}}', protocol_version=None)
```

**Comment by mattip** [comment]
I tried this with pip.

If I pip install `ray[core]` in a clean venv, I see it does not try to start the dashboard:
```
$ python3.10 -m venv /tmp/throwaway
$ source /tmp/throwaway/bin/activate
$ pip install "ray[core]"
$ python -c "import ray; ray.init()"
INFO worker.py:1664 -- Started a local Ray instance.
```

Now if I add `ray[default]` it installs many packages, none of them ray:
```
$ pip install "ray[default]"
...
Successfully installed 
aiohttp-3.9.1 
aiohttp-cors-0.7.0 
async-timeout-4.0.3 
blessed-1.20.0 
cachetools-5.3.2 
colorful-0.5.5 distlib-0.3.7 
google-api-core-2.14.0 
google-auth-2.23.4 
googleapis-common-protos-1.61.0 
gpustat-1.1.1 
grpcio-1.59.3 
multidict-6.0.4 
nvidia-ml-py-12.535.133 
opencensus-0.11.3 
opencensus-context-0.1.3 
platformdirs-3.11.0 
prometheus-client-0.19.0 
psutil-5.9.6 py-spy-0.3.14 
pyasn1-0.5.1 
pyasn1-modules-0.3.0 
pydantic-1.10.13 
rsa-4.9 
six-1.16.0 
smart-open-6.4.0 
typing-extensions-4.8.0 
virtualenv-20.21.0 
wcwidth-0.2.12 yarl-1.9.3

$ python -c "import ray; ray.init()"
INFO worker.py:1664 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265
```

Even though no new ray package has been installed, the dashboard is now active. What changed?

**Comment by mattip** [comment]
This is fixed for latest ray 2.8.1 in conda-forge/ray-packages-feedstock#133 where I merged the ray-dashboard package with ray-default so there is no longer a need to install something separate. I also added a passing test that this works: 
```
python -c "import ray; ray.init(include_dashboard=True)"
```

Thanks for the report. Please reopen or open a new issue if you notice problems with the conda-forge packaging.
